drivers/crypto/crypto_tc_shim.c

```diff
@@ -25,6 +25,61 @@ LOG_MODULE_REGISTER(tinycrypt);
 
 static struct tc_shim_drv_state tc_driver_state[CRYPTO_MAX_SESSION];
 
+static int do_ecb_encrypt(struct cipher_ctx *ctx, struct cipher_pkt *op,
+			  u8_t *iv)
+{
+	struct tc_shim_drv_state *data =  ctx->drv_sessn_state;
+
+	if (op->in_len != TC_AES_BLOCK_SIZE) {
+		LOG_ERR("ECB encryption only supported for a single "
+			"16 byte block");
+		return -EINVAL;
+	}
+
+	if (op->out_buf_max < TC_AES_BLOCK_SIZE) {
+		LOG_ERR("Output buffer too small for ECB operation");
+		return -EINVAL;
+	}
+
+	if (tc_aes_encrypt(op->out_buf,	op->in_buf,
+			   &data->session_key) == TC_CRYPTO_FAIL) {
+		LOG_ERR("TC internal error during ECB encryption");
+		return -EIO;
+	}
+
+	/* out_len is the same as in_len in ECB mode */
+	op->out_len = TC_AES_BLOCK_SIZE;
+
+	return 0;
+}
+
+static int do_ecb_decrypt(struct cipher_ctx *ctx, struct cipher_pkt *op,
+			  u8_t *iv)
+{
+	struct tc_shim_drv_state *data =  ctx->drv_sessn_state;
+
+	if (op->in_len != TC_AES_BLOCK_SIZE) {
+		LOG_ERR("ECB decryption only supported for a single "
+			"16 byte block");
+		return -EINVAL;
+	}
+
+	if (op->out_buf_max < TC_AES_BLOCK_SIZE) {
+		LOG_ERR("Output buffer too small for ECB operation");
+		return -EINVAL;
+	}
+
+	if (tc_aes_decrypt(op->out_buf, op->in_buf,
+			   &data->session_key) == TC_CRYPTO_FAIL) {
+		LOG_ERR("TC internal error during ECB decryption");
+		return -EIO;
+	}
+
+	op->out_len = TC_AES_BLOCK_SIZE;
+
+	return 0;
+}
+
```

Comment:
I think it'd be clear if you create a new function handler for ecb instead of re-using cbc handler. so in `include/crypto/cipher_structs.h"` you add:
```
+ typedef int (*ecb_op_t)(struct cipher_ctx *ctx, struct cipher_pkt *pkt);
struct cipher_ops {                                                             
                                                                                   
        enum cipher_mode cipher_mode;                                           
                                                                                
        union {                                                                 
                block_op_t      block_crypt_hndlr;                     
+                ecb_op_t        ecb_crypt_hndlr;                  
                cbc_op_t        cbc_crypt_hndlr;                                
                ctr_op_t        ctr_crypt_hndlr;                                
                ccm_op_t        ccm_crypt_hndlr;                                
                gcm_op_t        gcm_crypt_hndlr;                                
        };                                                                      
};    
```

Reply:
Hmm, I believe it’s supposed to use the block handler, isn’t it?

Reply:
Thanks for spotting that. I checked the API documentation and it seems like the intention is that block_crypt_hndlr is used for ECB and similar operations. This should be reflected in the updated PR.

Reply:
yep !